datanode/dehistory/snapshot/service_create_snapshot.go

```diff
@@ -88,9 +93,7 @@ func (b *Service) CreateSnapshot(ctx context.Context, chainID string, fromHeight
 		err = b.snapshotData(ctx, copyDataTx, dbMetaData, currentSnapshot, historySnapshot)
 		if err != nil {
 			b.log.Error("failed to snapshot data", logging.Error(err))
-			if b.config.PanicOnSnapshotCreationError {
-				panic(fmt.Sprintf("failed to snapshot data:%s", err))
-			}
+			panic(fmt.Sprintf("failed to snapshot data:%s", err))
```

Comment:
Could we use b.log.Panic() instead of bare panic? also you can combine this with the log previous to the panic.

Reply:
actually, this was b.log.panic once, I changed it to this as when it panic'ed using that it did notlog the error, which was a shame

Reply:
it does log the error when it panic, let's just use log.Panic really. it just log the error on the stdout and the stack trace on stderr, it's annoying I admit, but lets just do it with log.Panic as we do it like so everywhere